plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -60,23 +80,53 @@ protected void setup(Binder binder)
         newOptionalBinder(binder, Key.get(UserNameProvider.class, ForHiveMetastore.class))
                 .setDefault()
                 .toInstance(SIMPLE_USER_NAME_PROVIDER);
-
         binder.bind(Key.get(boolean.class, AllowHiveTableRename.class)).toInstance(true);
 
-        install(new ThriftMetastoreAuthenticationModule());
+        validateThriftHttpConfiguration();
     }
 
-    @Provides
-    @Singleton
-    @ThriftHiveWriteStatisticsExecutor
-    public ExecutorService createWriteStatisticsExecutor(ThriftMetastoreConfig hiveConfig)
+    private void validateThriftHttpConfiguration()
     {
-        return newFixedThreadPool(hiveConfig.getWriteStatisticsThreads(), threadsNamed("hive-thrift-statistics-write-%s"));
+        StaticMetastoreConfig staticMetastoreConfig = buildConfigObject(StaticMetastoreConfig.class);
+        ThriftHttpMetastoreConfig httpMetastoreConfig = buildConfigObject(ThriftHttpMetastoreConfig.class);
+        requireNonNull(staticMetastoreConfig.getMetastoreUris(), "metastoreUris is null");
+
+        boolean hasHttpMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                .anyMatch(uri -> "http".equalsIgnoreCase(uri.getScheme()));
+        boolean hasHttpsMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                .anyMatch(uri -> "https".equalsIgnoreCase(uri.getScheme()));
+
+        if (hasHttpMetastore || hasHttpsMetastore) {
+            if (hasHttpsMetastore && httpMetastoreConfig.getHttpBearerToken().isEmpty()) {
+                throw new IllegalStateException("'hive.metastore.http.client.bearer-token' must be set while using https metastore URIs in 'hive.metastore.uri'");
+            }
+            if (hasHttpMetastore && httpMetastoreConfig.getHttpBearerToken().isPresent()) {
+                throw new IllegalStateException("'hive.metastore.http.client.bearer-token' must not be set while using http metastore URIs in 'hive.metastore.uri'");
+            }
+            if (httpMetastoreConfig.getAuthenticationMode().isEmpty()) {
+                throw new IllegalStateException("'hive.metastore.http.client.authentication.type' must be set while using http/https metastore URIs in 'hive.metastore.uri'");
+            }
+        }
     }
 
     @PreDestroy
     public void shutdownsWriteStatisticExecutor(@ThriftHiveWriteStatisticsExecutor ExecutorService executor)
     {
         executor.shutdownNow();
     }
+
+    private record ThriftHiveMetastoreStatisticExecutorProvider(ThriftMetastoreConfig thriftMetastoreConfig)
```

Comment:
Please avoid putting config class to fields as it's mutable.

Reply:
I tried to remove this by passing in the int value of the number of write statistics threads, but then the binder fails with the error:
````
[ERROR]   TestHivePlugin.testThriftMetastore:70 » Creation Unable to create injector, see the following errors:

1) [Guice/JitDisabled]: Explicit bindings are required and Integer is not explicitly bound.
  at ThriftMetastoreModule$ThriftHiveMetastoreStatisticExecutorProvider.<init>(ThriftMetastoreModule.java:100)
      \_ for 1st parameter writeStatisticsThreads
  at ThriftMetastoreModule.setup(ThriftMetastoreModule.java:76)
      \_ installed by: HiveMetastoreModule -> ConditionalModule -> ThriftMetastoreModule
````

Reply:
You don't need to use record.

Reply:
Got it, I had originally changed it to record because the IDE suggested me. Let me revert it back to using private static class.